### PR TITLE
Presidio integration

### PR DESCRIPTION
**Pull request by RoeyBC**
# Description

Integrating presidio for PII detection and anonymization.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
@RoeyBC with the v0.10.0 release today, the package structure has changed for integrations 

You can create a new integration package under llama-index-integrations/postprocessors

More details here
https://github.com/run-llama/llama_index/blob/main/CONTRIBUTING.md#creating-a-new-integration-package

File: llama-index-integrations/postprocessor/llama-index-postprocessor-presidio/pyproject.toml
Comment (nerdai) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-presidio/pyproject.toml:33]: Can we also add section here for `[tool.llamahub]` similar to this [one](https://github.com/run-llama/llama_index/blob/f4bcb31f605f040d697dfe362d69616de47f2b48/llama-index-integrations/postprocessor/llama-index-postprocessor-cohere-rerank/pyproject.toml#L10). You can leave `contains_example` as `false`.

File: llama-index-integrations/postprocessor/llama-index-postprocessor-presidio/pyproject.toml
Comment (nerdai) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-presidio/pyproject.toml:20]: please add yourself as a `maintainer` like [here](https://github.com/run-llama/llama_index/blob/f4bcb31f605f040d697dfe362d69616de47f2b48/llama-index-packs/llama-index-packs-agents-llm-compiler/pyproject.toml#L26).

**Review by nerdai — COMMENTED** [review]
Thanks for the contribution @RoeyBC. High level looks pretty good. I left a few minor comments.

**Review by nerdai — APPROVED** [review]
Thanks @RoeyBC! LGTM. One thing I would suggest if not too much to add is an example notebook or script.

**Comment by RoeyBC** [comment]
@nerdai 100%!
I wasn't sure if it's supported before merging/publishing. Stay tuned :D

**Pull request merged**
